### PR TITLE
COMMIT0-57 api for generating project

### PR DESCRIPTION
**Pull request by liang-lillian**

### CODE REVIEW

File: internal/api/generate_api.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	Services          []Service    `json:"services"`
+}
+
+func generateProject(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	switch req.Method {
+	case "POST":
+		decoder := json.NewDecoder(req.Body)
+		var projectConfig ProjectConfiguration
+		err := decoder.Decode(&projectConfig)
```
Comment (liang-lillian) [inline internal/api/generate_api.go]: need to add validation for post request

File: internal/api/create_project.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	return rootDir
+}
+
+func GenerateArtifacts(projectConfig ProjectConfiguration) {
+
+	templates := packr.New("templates", "../templates")
+	t := templator.NewTemplator(templates)
+
+	err := os.Chdir(projectConfig.ProjectName) //cd into project
```
Comment (liang-lillian) [inline internal/api/create_project.go]: cd into project 
(could pass in project name to add to file path for each generate function)

File: internal/api/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Commit0 Api
+
+## Usage
+ - To run:
+ `go run internal/api/generate_api.go internal/api/create_project.go`
```
Comment (bmonkman) [inline internal/api/README.md]: Let's change this to an argument to the binary. `commit0 walkthrough` or `commit0 ui` or something

File: internal/generate/generate_helper.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/commitdev/commit0/internal/generate/react"
+	"github.com/commitdev/commit0/internal/templator"
+	"github.com/commitdev/commit0/internal/util"
+	"github.com/kyokomi/emoji"
+	"github.com/logrusorgru/aurora"
+	"log"
+	"sync"
+)
+
+func GenerateArtifactsHelper(t *templator.Templator, cfg *config.Commit0Config) {
```
Comment (liang-lillian) [inline internal/generate/generate_helper.go]: shared func for both cmd / api ` generate `

File: internal/api/create_project.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	if (err != nil) {
+		panic(err)
+	}
+	cfg := config.LoadConfig(util.ApiGeneratedYamlName)
```
Comment (bmonkman) [inline internal/api/create_project.go]: Let's just keep calling this `commit0.yaml`?

File: internal/templator/templator.go [hunk 6]
```diff
@@ -56,6 +57,7 @@ func NewTemplator(box *packr.Box) *Templator {
 		ProtoHealthTemplate:  NewSingleFileTemplator(box, "proto/health_proto.tmpl"),
 		ProtoServiceTemplate: NewSingleFileTemplator(box, "proto/service_proto.tmpl"),
 		Go:                   NewGoTemplator(box),
+		ApiCommit0:           NewSingleFileTemplator(box, "commit0/api_generated.tmpl"),
```
Comment (bmonkman) [inline internal/templator/templator.go]: The output of the ui process should just be a regular `commit0.yaml` using the same template

File: internal/api/create_project.go (hunk 2 above)
Comment (bmonkman) [reply internal/api/create_project.go]: Yeah you shouldn't need to chdir..

File: internal/api/README.md (hunk 3 above)
Comment (liang-lillian) [reply internal/api/README.md]: how do I do this? @bmonkman --ty!

Comment (bmonkman) [reply internal/api/README.md]: Yeah you would create a new file like `cmd/ui.go` which adds a new command

**Pull request merged**